### PR TITLE
[pmo] Eliminate dead flat namespace tuple numbering from PMOMemoryUseCollector.

### PR DESCRIPTION
**Pull request by gottesmm**
TLDR: This does not eliminate the struct/tuple flat namespace from Predictable
    Mem Opts. Just the tuple specific flat namespace code from PMOMemoryUseCollector
    that we were computing and then throwing away. I explain below in more detail.
    
    First note that this is cruft from when def-init and pmo were one pass. What we
    were doing here was maintaing a flattened tuple namespace while we were
    collecting uses in PMOMemoryUseCollector. We never actually used them for
    anything since we recomputed this information including information about
    structs in PMO itself! So this information was truly completely dead.
    
    This commit removes that and related logic and from a maintenance standpoint
    makes PMOMemoryUseCollector a simple visitor that doesn't have any real special
    logic in it beyond the tuple scalarization.

----

NOTE: I split this commit into two parts. 7175e17 only removes the flat namespace tuple numbering to ease review, while 7b7ccdc removes additional code that becomes dead. I did this to ease review.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Review by atrick — APPROVED** [review]
Awesome.
